Fails for me on macos 10.15:

```
make[3]: Entering directory '/private/tmp/nix-build-zbar-0.23.92.drv-0/source'
  CC       test/test_images.o
  CXX      test/test_cpp.o
  CC       test/test_decode-test_decode.o
  CXX      test/test_cpp_img.o
  CC       test/test_proc.o
  CC       test/test_jpeg.o
  CC       test/test_convert.o
test/test_jpeg.c:25:10: fatal error: 'argp.h' file not found
#include <argp.h>
         ^~~~~~~~
test/test_decode.c:24:10: fatal error: 'argp.h' file not found
#include <argp.h>
         ^~~~~~~~
```

ofborg seems to agree.

--- a/pkgs/tools/graphics/zbar/default.nix
+++ b/pkgs/tools/graphics/zbar/default.nix
@@ -24,6 +24,8 @@
 , libintl
 , libiconv
 , Foundation
+, bash
+, python3
 }:
 
 stdenv.mkDerivation rec {
@@ -66,6 +68,18 @@ stdenv.mkDerivation rec {
     qtx11extras
   ];
 
+  nativeCheckInputs = [
+    bash
+    python3
+  ];
+
+  # Note: postConfigure instead of postPatch in order to include some
+  # autoconf-generated files. The template files for the autogen'd scripts are
+  # not chmod +x, so patchShebangs misses them.
+  postConfigure = ''
+    patchShebangs test
+  '';
+
   # Disable assertions which include -dev QtBase file paths.
   env.NIX_CFLAGS_COMPILE = "-DQT_NO_DEBUG";
 
@@ -82,6 +96,8 @@ stdenv.mkDerivation rec {
     "--without-gtk"
     "--without-qt"
   ]);
+
+  doCheck = !stdenv.isDarwin;
 
   dontWrapQtApps = true;
   dontWrapGApps = true;

--- a/pkgs/tools/graphics/zbar/default.nix
+++ b/pkgs/tools/graphics/zbar/default.nix
@@ -28,7 +28,7 @@
 
 stdenv.mkDerivation rec {
   pname = "zbar";
-  version = "0.23.90";
+  version = "0.23.92";
 
   outputs = [ "out" "lib" "dev" "doc" "man" ];
 
@@ -36,7 +36,7 @@ stdenv.mkDerivation rec {
     owner = "mchehab";
     repo = "zbar";
     rev = version;
-    sha256 = "sha256-FvV7TMc4JbOiRjWLka0IhtpGGqGm5fis7h870OmJw2U=";
+    sha256 = "sha256-VhVrngAX7pXZp+szqv95R6RGAJojp3svdbaRKigGb0w=";
   };
 
   nativeBuildInputs = [

--- a/pkgs/tools/graphics/zbar/default.nix
+++ b/pkgs/tools/graphics/zbar/default.nix
@@ -26,6 +26,7 @@
 , Foundation
 , bash
 , python3
+, argp-standalone
 }:
 
 stdenv.mkDerivation rec {
@@ -73,6 +74,10 @@ stdenv.mkDerivation rec {
     python3
   ];
 
+  checkInputs = lib.optionals stdenv.isDarwin [
+    argp-standalone
+  ];
+
   # Note: postConfigure instead of postPatch in order to include some
   # autoconf-generated files. The template files for the autogen'd scripts are
   # not chmod +x, so patchShebangs misses them.
@@ -97,7 +102,11 @@ stdenv.mkDerivation rec {
     "--without-qt"
   ]);
 
-  doCheck = !stdenv.isDarwin;
+  doCheck = true;
+
+  preCheck = lib.optionalString stdenv.isDarwin ''
+    export NIX_LDFLAGS="$NIX_LDFLAGS -largp"
+  '';
 
   dontWrapQtApps = true;
   dontWrapGApps = true;

